nts-pool-ke/src/cli.rs

```diff
@@ -0,0 +1,190 @@
+use crate::daemon_tracing::LogLevel;
+use std::path::PathBuf;
+use std::str::FromStr;
+
+const USAGE_MSG: &str = "\
+usage: nts-pool-ke [-c PATH] [-l LOG_LEVEL]
+       nts-pool-ke -h
+       nts-pool-ke -v";
+
+const DESCRIPTOR: &str = "ntp-daemon - synchronize system time";
+
+const HELP_MSG: &str = "Options:
+  -c, --config=PATH             change the config .toml file
+  -l, --log-level=LOG_LEVEL     change the log level
+  -h, --help                    display this help text
+  -v, --version                 display version information";
+
+pub fn long_help_message() -> String {
+    format!("{DESCRIPTOR}\n\n{USAGE_MSG}\n\n{HELP_MSG}")
+}
+
+#[derive(Debug, Default)]
+pub(crate) struct NtsPoolKeOptions {
+    /// Path of the configuration file
+    pub config: Option<PathBuf>,
+    /// Level for messages to display in logs
+    pub log_level: Option<LogLevel>,
+    help: bool,
+    version: bool,
+    pub action: NtsPoolKeAction,
+}
+
+pub enum CliArg {
+    Flag(String),
+    Argument(String, String),
+    Rest(Vec<String>),
+}
+
+impl CliArg {
+    pub fn normalize_arguments<I>(
+        takes_argument: &[&str],
+        takes_argument_short: &[char],
+        iter: I,
+    ) -> Result<Vec<Self>, String>
+    where
+        I: IntoIterator<Item = String>,
+    {
+        // the first argument is the sudo command - so we can skip it
```

Comment:
This comment seems incorrect

Reply:
hah, that' s incorrect in the original too then